diff --git a/core/modules/migrate/src/Plugin/Migration.php b/core/modules/migrate/src/Plugin/Migration.php
index 86294b5..3b666a4 100644
--- a/core/modules/migrate/src/Plugin/Migration.php
+++ b/core/modules/migrate/src/Plugin/Migration.php
@@ -662,7 +662,7 @@ public function setTrackLastImported($track_last_imported) {
    * {@inheritdoc}
    */
   public function getMigrationDependencies() {
-    return $this->migration_dependencies + ['required' => [], 'optional' => []];
+    return ($this->migration_dependencies ?: []) + ['required' => [], 'optional' => []];
   }
 
   /**
diff --git a/core/modules/migrate/tests/src/Kernel/Plugin/MigrationTest.php b/core/modules/migrate/tests/src/Kernel/Plugin/MigrationTest.php
index a380dac..35c4da7 100644
--- a/core/modules/migrate/tests/src/Kernel/Plugin/MigrationTest.php
+++ b/core/modules/migrate/tests/src/Kernel/Plugin/MigrationTest.php
@@ -27,4 +27,16 @@ public function testGetProcessPlugins() {
     $this->assertEquals([], $migration->getProcessPlugins([]));
   }
 
+  /**
+   * Tests Migration::getMigrationDependencies()
+   *
+   * @covers ::getMigrationDependencies
+   */
+  public function testGetMigrationDependencies() {
+    $migration = \Drupal::service('plugin.manager.migration')->createStubMigration([
+        'migration_dependencies' => NULL
+    ]);
+    $this->assertNotEmpty($migration->getMigrationDependencies(), 'Migration dependencies is not empty');
+  }
+
 }

